creds_generator.go

```diff
@@ -0,0 +1,65 @@
+package authboss
+
+import (
+	"crypto/rand"
+	"crypto/sha512"
+	"encoding/base64"
+	"io"
+)
+
+type CredsGenerator interface {
```

Comment:
All these interfaces should be documented in Godoc.

Reply:
Thoughts on renaming CredsGenerator -> OneTimeTokenGenerator?

Credentials is a bit overloaded as a noun.